src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  smodel = get_refined_model(strian)
+  tmodel = get_background_model(ttrian)
+  a = get_model(smodel) === tmodel  # It is fundamentally the same model
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  return is_change_possible(ttrian,strian)
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  @check is_change_possible(strian,ttrian)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_model(smodel) === get_parent(tmodel)
+  b = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  a || b ? ttrian : strian
+end
+
+function Geometry.best_target(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return strian
+end
+
+function Geometry.best_target(strian::T,ttrian::RefinedTriangulation) where {T <: Triangulation}
+  @check is_change_possible(strian,ttrian)
+  return ttrian
+end
+
+
+"""
+  Given a RefinedTriangulation and a CellField defined on the parent(coarse) mesh, 
+  returns an equivalent CellField on the fine mesh.
+"""
+function change_domain_c2f(f_coarse, ftrian::RefinedTriangulation{Dc,Dp}) where {Dc,Dp}
+  model  = get_refined_model(ftrian)
+  glue   = get_glue(model)
+  if (num_cells(ftrian) != 0)
+    # Coarse field but with fine indexing, i.e 
+    #   f_f2c[i_fine] = f_coarse[coarse_parent(i_fine)]
+    fcell_to_ccell = glue.f2c_faces_map[Dc+1]
+    m = Reindex(get_data(f_coarse))
+    f_f2c = lazy_map(m,fcell_to_ccell)
+
+    # Fine to coarse coordinate map: x_coarse = Φ^(-1)(x_fine)
+    ref_coord_map = get_f2c_ref_coordinate_map(glue)
+
+    # Final map: f_fine(x_fine) = f_f2c ∘ Φ^(-1)(x_fine) = f_coarse(x_coarse)
+    f_fine = lazy_map(∘,f_f2c,ref_coord_map)
+    return GenericCellField(f_fine,ftrian,ReferenceDomain())
+  else
+    f_fine = Fill(Gridap.Fields.ConstantField(0.0),num_cells(ftrian))
+    return GenericCellField(f_fine,ftrian,ReferenceDomain())
+  end
+end
+
+function CellData.change_domain(a::CellField,ttrian::RefinedTriangulation)
+  strian = get_triangulation(a)
+  if strian === ttrian
```

Comment:
I think this function does not work properly if, e.g., `get_triangulation(CellField)` is a view of `ttrian`.

I think this is related to a previous comment, and also does the potential solution.

Reply:
Resolved by [0e3c6eb](https://github.com/gridap/Gridap.jl/pull/838/commits/0e3c6eb9cdc1b3dd995a65f7732f940ee853e28b)